pkg/operator/configobservation/auth/auth_serviceaccountissuer.go

```diff
@@ -79,52 +93,54 @@ func observedConfig(existingConfig map[string]interface{},
 		return existingConfig, append(errs, err)
 	}
 
-	newActiveIssuer = getActiveServiceAccountIssuer(operator.Status.ServiceAccountIssuers)
-	if err := checkIssuer(newActiveIssuer); err != nil {
+	// desiredActiveIssuer is the desired service account issuer set in KAS operator status
+	// This apiServerArgumentValue is being synced using serviceaccountissuer controller.
+	desiredActiveIssuer = getActiveServiceAccountIssuer(operator.Status.ServiceAccountIssuers)
+	// if desired active issuer is not set (the serviceaccountissuer for some reason has not defaulted it)
+	// then make sure, we default it here, because we have to set the jwks-uri correctly.
+	if existingConfigIssuer == defaultServiceAccountIssuerValue && len(desiredActiveIssuer) == 0 {
```

Comment:
`if existingConfigIssuer != defaultServiceAccountIssuerValue` but `desiredActiveIssuer == ""` we don't set `desiredActiveIssuer=defaultServiceAccountIssuerValue`?

Reply:
i struggle how that can happen other than somebody manually tweaking the KAS config :-)